src/dali/plugin/input/csv/crypto_com_app.py

```diff
@@ -0,0 +1,465 @@
+# 10- reimbursement_reverted: is a cancellation of a previous reward given
+# 11- referral_card_cashback: is cashback from using the debit card
+# 12- mco_stake_reward: is crypto paid to me from locking into the staking program
+# 13- crypto_earn_interest_paid: is crypto paid to me from looking into the earn program
+# 14- crypto_purchase: is a crypto buy with fiat
+# 15- crypto_exchange: exchange between crypto
+# 16- crypto_withdrawal: sending crypto to another wallet
+# 17- crypto_viban_exchange: selling crypto to get fiat
+# 18- crypto_earn_program_withdrawn: is unlocking money for use
+# 19- card_cashback_reverted: is a cancellation of a previous reward given
+
+# IN-transactions describe crypto flowing in (e.g. airdrop, buy, hard fork, income, interest, mining, staking, wages)
+_IN_TYPE_MAPPING = {
+    "referral_gift": Keyword.INCOME.value,  # Signup bonus or promotions
+    "viban_purchase": Keyword.BUY.value,  # Buying crypto with fiat
+    "reimbursement": Keyword.INCOME.value,  # Cashback from services like Netflix
+    "referral_card_cashback": Keyword.INCOME.value,  # Cashback from using the debit card
+    "mco_stake_reward": Keyword.STAKING.value,  # Rewards from staking MCO
+    "crypto_earn_interest_paid": Keyword.INTEREST.value,  # Interest paid from the earn program
+    "crypto_purchase": Keyword.BUY.value,  # Buying crypto with fiat
+}
+
+# OUT-transactions describe crypto flowing out (e.g. donate, fee, gift, sell)
+_OUT_TYPE_MAPPING = {
+    "crypto_viban_exchange": Keyword.SELL.value,
+    "card_top_up": Keyword.SELL.value,
+    "reimbursement_reverted": Keyword.SELL.value,
+    "card_cashback_reverted": Keyword.SELL.value,
+}
+
+# INTRA-transactions describe crypto moving across accounts
+_INTRA_MAPPING = {"crypto_withdrawal", Keyword.MOVE.value}
+
+# Reversal transactions that could be removed from the records
+_REVERSAL_MAPPING = {"reimbursement_reverted", "card_cashback_reverted"}
+
+# Exchange between currencies to be converted into a sell and buy orders.
+_TRANSACTION_MAPPING = {"crypto_exchange"}
+
+# Ignored transactions that do not need to be recorded
+_IGNORED_MAPPING = {
+    "viban_deposit_precredit",
+    "viban_deposit_precredit_repayment",
+    "lockup_lock",
+    "lockup_unlock",
+    "crypto_earn_program_created",
+    "crypto_earn_program_withdrawn",
+}
+
+
+class CryptoComAppTransaction:
+    def __init__(
+        self,
+        raw_data: str,
+        time: str,
+        description: str,
+        currency: str,
+        amount: str,
+        native_currency: str,
+        native_amount: str,
+        native_amount_usd: str,
+        transaction_kind: str,
+        to_currency: Optional[str] = None,
+        to_amount: Optional[str] = None,
+        transaction_hash: Optional[str] = None,
+        notes: Optional[str] = None,
+    ):
+        self.raw_data = raw_data
+        self.timestamp = time
+        self.abs_float_amount = abs(float(amount))
```

Comment:
Use RP2Decimal instead of float throughout the file to avoid precision problems.